pkgs/by-name/wa/wamrc/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  llvmPackages,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wamrc";
+  version = "2.3.1";
+
+  src = fetchFromGitHub {
+    owner = "bytecodealliance";
+    repo = "wasm-micro-runtime";
+    tag = "WAMR-${finalAttrs.version}";
+    hash = "sha256-Rhn26TRyjkR30+zyosfooOGjhvG+ztYtJVQlRfzWEFo=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    llvmPackages.llvm
+  ];
+
+  cmakeFlags =
+    lib.optionals stdenv.hostPlatform.isDarwin [
+      "-DCMAKE_OSX_DEPLOYMENT_TARGET=${stdenv.hostPlatform.darwinSdkVersion}"
+    ]
+    ++ [
+      "-DWAMR_BUILD_WITH_CUSTOM_LLVM=ON"
+      "-DLLVM_DIR=${llvmPackages.llvm.dev}/lib/cmake/llvm"
+    ];
+
```

Comment:
Nit: add nix-update-script, for how to overcome tag version seet #432501
LGTM otherwise